src/bin/gpcheckhdfs/gpcheckhdfs.c

```diff
@@ -110,6 +114,12 @@ int main(int argc, char * argv[]) {
         krb_keytabfile = argv[5];
     }
 
+    //get tde key name param
+    for (int i = 1; i < argc; i++) {
+        if (strcmp(argv[i], "--with-tde") == 0 && i+1 < argc)
+            tde_keyname = argv[i+1];
```

Comment:
why not use `argv[6]` as the `tde_keyname`?
because the params before it may be empty...